### DISCUSSION TITLE
How to migrate from Oracle Java to Microsoft OpenJDK

### DISCUSSION THREAD

**Discussion by maxwell2k19** [microsoft/openjdk#381] [Q&A] [answered]
Hi All

I have many servers which have outdated oracle Java. I want to uninstall oracle java and install Microsoft openjdk. Please guide me how to migrate the configuration from oracle java to Microsoft OpenJDK.

**Comment by karianna** [comment, marked as answer]
@maxwell2k19 Which version of Java are you currently using?

**Reply by maxwell2k19** [reply]
Path              : C:\Program Files\Java\jre1.8.0_331\
  Installed version : 1.8.0_331

**Reply by karianna** [reply]
OK, so you have two options:

1. Migrate to Eclipse Temurin 8 binaries (Java 8 compatible) - Eclipse Temurin is the distribution that Microsoft is involved in for Java 8 (although no commercial support is given unless you are on Azure).  Get this from adoptium.net and look at the guide https://adoptium.net/docs/migration/

2. Migrate to Microsoft Build of OpenJDK 11 - (although no commercial support is given unless you are on Azure).  Get this from microsoft.com/openjdk and look at the guides: https://learn.microsoft.com/en-us/java/openjdk/reasons-to-move-to-java-11 and https://learn.microsoft.com/en-us/java/openjdk/transition-from-java-8-to-java-11
